ArcGISRuntimeSDKQt_CppSamples/Search/ReverseGeocodeOnline/ReverseGeocodeOnline.cpp

```diff
@@ -0,0 +1,153 @@
+// [WriteFile Name=ReverseGeocodeOnline, Category=Search]
+// [Legal]
+// Copyright 2020 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "ReverseGeocodeOnline.h"
+
+#include "CalloutData.h"
+#include "Graphic.h"
+#include "GraphicsOverlay.h"
+#include "LocatorTask.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "PictureMarkerSymbol.h"
+#include "SimpleRenderer.h"
+
+#include <QDir>
+#include <QUrl>
+
+using namespace Esri::ArcGISRuntime;
+
+namespace
+{
+// helper method to get cross platform data path
+QString defaultDataPath()
+{
+  QString dataPath;
+
+#ifdef Q_OS_ANDROID
+  dataPath = "/sdcard";
+#elif defined Q_OS_IOS
+  dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+#else
+  dataPath = QDir::homePath();
+#endif
+
+  return dataPath;
+}
+
+const QUrl url("http://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer");
+const QUrl pinUrl(defaultDataPath() + "/ArcGIS/Runtime/Data/symbol/pin.png");
+} // namespace
+
+ReverseGeocodeOnline::ReverseGeocodeOnline(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(Basemap::imageryWithLabels(this), this))
+{
+  m_locatorTask = new LocatorTask(url, this);
+}
+
+ReverseGeocodeOnline::~ReverseGeocodeOnline() = default;
+
+void ReverseGeocodeOnline::configureGraphic()
+{
+  m_graphicsOverlay = new GraphicsOverlay(this);
+
+  PictureMarkerSymbol* pinSymbol = new PictureMarkerSymbol(pinUrl, this);
+  pinSymbol->setHeight(72);
+  pinSymbol->setWidth(19);
+  SimpleRenderer* simpleRenderer = new SimpleRenderer(pinSymbol, this);
+
+  m_graphicsOverlay->setRenderer(simpleRenderer);
+  m_graphicsOverlay->graphics()->append(new Graphic(this));
+  m_graphic = m_graphicsOverlay->graphics()->at(0);
+}
+
+void ReverseGeocodeOnline::getAddress()
+{
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](QMouseEvent& e)
+  {
+    const Point clickedLocation = m_mapView->screenToLocation(e.x(), e.y());
+    ReverseGeocodeParameters* reverseGeocodeParameters = new ReverseGeocodeParameters();
```

Comment:
missing a parent? If our ReverseGeocodeParameters are a value not a QObject type then they should not be allocated like this (will leak)